File: tests/transpiler/x/pl/right_join.pl
:- initialization(main).
:- style_check(-singleton).

main :-
    Customers = [map{id: 1, name: "Alice"}, map{id: 2, name: "Bob"}, map{id: 3, name: "Charlie"}, map{id: 4, name: "Diana"}],
    Orders = [map{id: 100, customerId: 1, total: 250}, map{id: 101, customerId: 2, total: 125}, map{id: 102, customerId: 1, total: 300}],
    Result = [map{customerName: "Alice", order: map{id: 100, customerId: 1, total: 250}}, map{customerName: "Bob", order: map{id: 101, customerId: 2, total: 125}}, map{customerName: "Alice", order: map{id: 102, customerId: 1, total: 300}}],
    writeln("--- Right Join using syntax ---"),
    Entry = map{customerName: "Alice", order: map{id: 100, customerId: 1, total: 250}},
write("Customer"), write(' '), write("Alice"), write(' '), write("has order"), write(' '), get_dict(id, get_dict(order, Entry, R), V5), write(V5), write(' '), write("- $"), write(' '), get_dict(total, get_dict(order, Entry, R), V6), writeln(V6),
    Entry1 = map{customerName: "Bob", order: map{id: 101, customerId: 2, total: 125}},
write("Customer"), write(' '), write("Bob"), write(' '), write("has order"), write(' '), get_dict(id, get_dict(order, Entry1, R), V7), write(V7), write(' '), write("- $"), write(' '), get_dict(total, get_dict(order, Entry1, R), V8), writeln(V8),
    Entry2 = map{customerName: "Alice", order: map{id: 102, customerId: 1, total: 300}},
write("Customer"), write(' '), write("Alice"), write(' '), write("has order"), write(' '), get_dict(id, get_dict(order, Entry2, R), V9), write(V9), write(' '), write("- $"), write(' '), get_dict(total, get_dict(order, Entry2, R), V10), writeln(V10).

File: archived/tests/transpiler/x/fs/right_join.fs
// Generated 2025-07-21 18:37 +0700

type Anon1 = {
    mutable id: int
    mutable name: string
}
type Anon2 = {
    mutable id: int
    mutable name: string
}
type Anon3 = {
    mutable id: int
    mutable customerId: int
    mutable total: int
}
type Anon4 = {
    mutable id: int
    mutable customerId: int
    mutable total: int
}
type Anon5 = {
    mutable customerName: obj
    mutable order: obj
}
type Anon6 = {
    mutable customerName: obj
    mutable order: obj
}
let customers: Anon2 list = [{ id = 1; name = "Alice" }; { id = 2; name = "Bob" }; { id = 3; name = "Charlie" }; { id = 4; name = "Diana" }]
let orders: Anon4 list = [{ id = 100; customerId = 1; total = 250 }; { id = 101; customerId = 2; total = 125 }; { id = 102; customerId = 1; total = 300 }]
let result: Anon6 list = [ for c in customers do for o in orders do if (o.customerId) = (c.id) then yield { customerName = c.name; order = o } ]
printfn "%s" (string "--- Right Join using syntax ---")
for entry in result do
if entry.order then
printfn "%s" (String.concat " " [string "Customer"; string (entry.customerName); string "has order"; string (entry.order.id); string "- $"; string (entry.order.total)])
else
printfn "%s" (String.concat " " [string "Customer"; string (entry.customerName); string "has no orders"])
